ci_scripts/create_doc.sh

```diff
@@ -0,0 +1,37 @@
+set -euo pipefail
+
+# Check if DOCPUSH is set
+if ! [[ -z ${DOCPUSH+x} ]]; then
+
+    if [[ "$DOCPUSH" == "true" ]]; then
+
+        # install documentation building dependencies
+        pip install sphinx sphinx-gallery sphinx_bootstrap_theme
+        # $1 is the branch name
+        # $2 is the global variable where we set the script status
+
+        if ! [[ $1 = "master" ]]; then
```

Comment:
Would it make sense to move this after the actual documentation building to check that it works well?

Reply:
Hmm, If I am not mistaken, even if we move the if condition after the documentation building, if the documentation building fails, it will not fail the travis build. To check that the documentation is being built successfully, one would have to investigate the travis log and check the output of the documentation building command.  
Frankly, this would be useful only to detect that the documentation is not being built on other branches, even though the documentation is not deployed. This would only help see a possible issue early, since it will still be visible when the branch is merged into master. I can change it if you still want it.

Nicely observed, I never thought about deleting the gh-pages. I can add the .nojekyll to the openml-python repo and the above change, if you want to proceed with the changed order.

Reply:
Won't `set -e` mean that the documentation build fail is propagated? Especially with `SPHINXOPTS=-W`?

Reply:
> Won't `set -e` mean that the documentation build fail is propagated? Especially with `SPHINXOPTS=-W`?

Yes it does. However from the travis docs:

`The exit code of after_success, after_failure, after_script, after_deploy and subsequent stages do not affect the build result. However, if one of these stages times out, the build is marked as failed.`

https://docs.travis-ci.com/user/job-lifecycle/#breaking-the-build

Reply:
>  This would only help see a possible issue early

That's what I would like to have to make sure that everything we merge into master doesn't break the docs. How about moving the docs building to the script section as a second command (as in https://docs.travis-ci.com/user/job-lifecycle/#customizing-the-build-phase)?